src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -439,6 +443,11 @@ internal static void CheckParameterModifiers(BaseParameterSyntax parameter, Bind
                 switch (modifier.Kind())
                 {
                     case SyntaxKind.ThisKeyword:
+                        Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod, diagnostics);
+
+                        if (seenRef || seenIn)
+                            Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods, diagnostics);
```

Comment:
I think both `ref this` and `this ref` are supported, but it looks like this will only catch the case where `ref``/`in` comes first.

Reply:
=-o

Reply:
I have learned something new today!

Reply:
Fixed.